runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -251,14 +300,16 @@ public final void appendSummaryHtml(PrintWriter writer) {
         streamClosed.get());
   }
 
-  // Don't require synchronization on stream, see the appendSummaryHtml comment.
+  /**
+   * @implNote Don't require synchronization on stream, see the {@link
+   *     #appendSummaryHtml(PrintWriter)} comment.
+   */
   protected abstract void appendSpecificHtml(PrintWriter writer);
 
   @Override
-  public final synchronized void halfClose() {
-    // Synchronization of close and onCompleted necessary for correct retry logic in onNewStream.
+  public final void halfClose() {
     clientClosed.set(true);
-    requestObserver().onCompleted();
+    requestObserver.onCompleted();
```

Comment:
let me try to run a pipeline w synchronization

previously we were calling relying on this to shutdown, so we had removed synchronization but since we arent anymore it could be ok

Reply:
ran ok in the test pipeline

done.